core/src/commonMain/kotlin/com/powersync/PsSqlDriver.kt

```diff
@@ -1,46 +1,49 @@
 package com.powersync
 
 import app.cash.sqldelight.db.SqlDriver
+import com.powersync.utils.AtomicMutableSet
 import kotlinx.coroutines.CoroutineScope
-import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.async
 import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.SharedFlow
 import kotlinx.coroutines.flow.asSharedFlow
-import kotlinx.coroutines.flow.filter
-import kotlinx.coroutines.flow.map
 import kotlinx.coroutines.launch
 
 internal class PsSqlDriver(
     private val driver: SqlDriver,
     private val scope: CoroutineScope,
 ) : SqlDriver by driver {
     // MutableSharedFlow to emit batched table updates
-    private val tableUpdatesFlow = MutableSharedFlow<List<String>>(replay = 0)
+    private val tableUpdatesFlow = MutableSharedFlow<Set<String>>(replay = 0)
 
     // In-memory buffer to store table names before flushing
-    private val pendingUpdates = mutableSetOf<String>()
+    private val pendingUpdates = AtomicMutableSet<String>()
```

Comment:
I generally subscribe to the idea that the use synchronized collections is an anti-pattern because it's still possible to have concurrency issues with then when not used properly. That's not the case here, but I still think we should remove methods from `AtomicMutableSet` that we're not using.

So perhaps we can remove the unused `remove`, `clear` and `contains` methods? And IMO the `clear` parameter on `toSet` can be removed too, maybe just call the method `toSetAndClear`. And then we could rename the class to e.g. `BatchBuilder` or something, but that's a nit and might be going a bit far.

Reply:
Good suggestion! I've made some of these changes when updating to a synchronised object implementation.